src/internal/util/EmptyError.ts

```diff
@@ -9,9 +9,11 @@
  * @class EmptyError
  */
 export class EmptyError extends Error {
+
+  public readonly name: 'EmptyError' = 'EmptyError';
```

Comment:
I'm surprised why is this notation `name: 'EmptyError' = 'EmptyError';` required? It think using just:

    public readonly name = 'EmptyError';

should be enough for TypeScript's to type guard to enforce only string declared as `readonly` anyway. [Here the same thing is defined](https://github.com/ReactiveX/rxjs/pull/3678/files#diff-5446941a3fbe64a64353c2fa505ded36R493) as `name: string = 'AjaxTimeoutError';`.

Reply:
@martinsik You're right, we only need to do `public readonly name = 'EmptyError';` to enforce.  

The `AjaxTimeoutError` was implementing the parent class `AjaxError`'s `name` property so that was why it was a string.  I added a new intersection type `type AjaxErrors = 'AjaxError' | 'AjaxTimeoutError';` Let me know what you think.